### PR TITLE
Restore Azure build on ARM

### PR DESCRIPTION
**Pull request by rschu1ze**
Reverts #50770

As a side effect, this PR also enables Hive on ARM.

### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

**Comment by robot-ch-test-poll2** [comment]
This is an automated comment for commit 488ed3a99b7e1627f03cca28a8899d13236eeb23 with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/51288/488ed3a99b7e1627f03cca28a8899d13236eeb23/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

File: contrib/libhdfs3-cmake/CMakeLists.txt [hunk 1]
```diff
@@ -1,11 +1,11 @@
-if(NOT ARCH_AARCH64 AND NOT OS_FREEBSD AND NOT APPLE AND NOT ARCH_PPC64LE AND NOT ARCH_S390X)
+if(NOT OS_FREEBSD AND NOT APPLE AND NOT ARCH_PPC64LE AND NOT ARCH_S390X)
```
Comment (nikitamikhaylov) [inline contrib/libhdfs3-cmake/CMakeLists.txt:1]: What about supporting HDFS on Mac? Probably it can be just enabled and it will work?

**Review by nikitamikhaylov — APPROVED** [review]

File: contrib/libhdfs3-cmake/CMakeLists.txt (hunk 1 above)
Comment (rschu1ze) [reply contrib/libhdfs3-cmake/CMakeLists.txt:1]: Doable but a bit more complicated (especially curl on Mac is tricky). [Another PR](https://github.com/ClickHouse/ClickHouse/pull/51276) attempts this.

Comment (nikitamikhaylov) [reply contrib/libhdfs3-cmake/CMakeLists.txt:1]: Yeah, missed it, sorry

**Pull request merged**